### DISCUSSION TITLE
How to construct a label using a loop?

### DISCUSSION THREAD

**Discussion by xbmbmx** [typst/typst#3271] [Q&A] [answered]
My MWE has problems, but the general question is clear from my attempt I think:

```typ
#table(
  columns: (auto, auto),
  align: center,
  for x in range(1, 100) {
    [#x], [#(100-x)]
  }
)
```

So:
1. How do I use this loop to generate multiple rows based on the loop iterator value?
2. How do I do the substraction in the second field on each row? (this is pseudocode)

I've been looking around a bit but I can't seem to find a straight method to achieve this.

**Comment by laurmaedje** [comment, marked as answer]
You have two options:

```typ
#table(
  columns: (auto, auto),
  align: center,
  ..for x in range(1, 100) {
    ([#x], [#(100-x)])
  }
)
```

This works because the individual arrays of each for loop iteration join into one big array, which can be given to the table function item-by-item with `..`.

Or:

```typ
#table(
  columns: (auto, auto),
  align: center,
  ..range(1, 100).map(x => ([#x], [#(100-x)])).flatten()
)
```

The subtraction looks correct.

**Reply by xbmbmx** [reply]
Perfect! Thank you.
